packages/game-passes/test/GamePasses.test.ts

```diff
@@ -0,0 +1,1251 @@
+        );
+
+      const tokenConfig = await sandboxPasses.tokenConfigs(NEW_TOKEN_ID);
+      expect(tokenConfig.isConfigured).to.be.true;
+      expect(tokenConfig.transferable).to.be.true;
+      expect(tokenConfig.maxSupply).to.equal(200);
+      expect(tokenConfig.maxPerWallet).to.equal(20);
+      expect(tokenConfig.metadata).to.equal('ipfs://QmNewToken');
+      expect(tokenConfig.treasuryWallet).to.equal(user1.address);
+    });
+
+    it('should not allow non-admin to configure a token', async function () {
+      await expect(
+        sandboxPasses
+          .connect(user1)
+          .configureToken(3, true, 100, 10, 'metadata', ZERO_ADDRESS),
+      ).to.be.revertedWithCustomError(
+        sandboxPasses,
+        'AccessControlUnauthorizedAccount',
+      );
+    });
+
+    it("should not allow configuring a token that's already configured", async function () {
+      await expect(
+        sandboxPasses
+          .connect(admin)
+          .configureToken(TOKEN_ID_1, true, 100, 10, 'metadata', ZERO_ADDRESS),
+      ).to.be.revertedWithCustomError(sandboxPasses, 'TokenAlreadyConfigured');
+    });
+
+    it('should allow updating token configuration', async function () {
+      await sandboxPasses.connect(admin).updateTokenConfig(
+        TOKEN_ID_1,
+        200, // new max supply
+        15, // new max per wallet
+        'ipfs://QmUpdated',
+        user2.address,
+      );
+
+      const tokenConfig = await sandboxPasses.tokenConfigs(TOKEN_ID_1);
+      expect(tokenConfig.maxSupply).to.equal(200);
+      expect(tokenConfig.maxPerWallet).to.equal(15);
+      expect(tokenConfig.metadata).to.equal('ipfs://QmUpdated');
+      expect(tokenConfig.treasuryWallet).to.equal(user2.address);
+    });
+
+    it('should not allow decreasing max supply below current supply', async function () {
+      // First mint some tokens
+      await sandboxPasses
+        .connect(admin)
+        .adminMint(user1.address, TOKEN_ID_1, 50);
+
+      // Try to update max supply to below current supply
+      await expect(
+        sandboxPasses
+          .connect(admin)
+          .updateTokenConfig(TOKEN_ID_1, 40, 10, TOKEN_METADATA, ZERO_ADDRESS),
+      ).to.be.revertedWithCustomError(
+        sandboxPasses,
+        'MaxSupplyBelowCurrentSupply',
+      );
+    });
+
+    it('should allow setting transferability', async function () {
+      // Change transferable token to non-transferable
+      await sandboxPasses.connect(admin).setTransferable(TOKEN_ID_1, false);
+
+      const tokenConfig = await sandboxPasses.tokenConfigs(TOKEN_ID_1);
+      expect(tokenConfig.transferable).to.be.false;
+
+      // Change non-transferable token to transferable
+      await sandboxPasses.connect(admin).setTransferable(TOKEN_ID_2, true);
+
+      const tokenConfig2 = await sandboxPasses.tokenConfigs(TOKEN_ID_2);
+      expect(tokenConfig2.transferable).to.be.true;
+    });
+
+    it('should allow updating transfer whitelist', async function () {
```

Comment:
missing opposite test if not admin

Reply:
Thank you, updated in https://github.com/thesandboxgame/sandbox-smart-contracts/pull/1619/commits/841b26173a36c207db241ddf6603ad8ed2903abe